code/game/objects/structures/beacon.dm

```diff
@@ -0,0 +1,45 @@
+/obj/structure/strangebeacon
+	icon = 'icons/obj/machines/excelsior/objects.dmi'
+	icon_state = "strange_beacon"
+	var/nosignal = 0
+
+/obj/structure/strangebeacon/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		playsound(loc, 'sound/machines/twobeep.ogg', 50, 1)	//Plays a beep
+		nosignal = 1
+
+/obj/structure/strangebeacon/bots //who you gonna call
+/obj/structure/strangebeacon/bots/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		var/counter = 0
+		var/counterfinish = rand(1,2)
+
+		while(counter < counterfinish)
+			counter++
+			new /mob/living/bot/miningonestar/resources/agressive ( locate( get_step(src, pick(NORTH, WEST, EAST, SOUTH) ) ))
+		nosignal = 1
+
+/obj/structure/strangebeacon/pods
+/obj/structure/strangebeacon/pods/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		var/drop_x = src.x-2
+		var/drop_y = src.y-2
+		var/drop_z = src.z
+		var/drop_type = pick(supply_drop_random_loot_types())
+
+		spawn(rand(100,300))
```

Comment:
Spawns should be avoided, especially for long-timer stuff like that https://github.com/Baystation12/Baystation12/wiki/Timers
e.g. `addtimer(CALLBACK(src, .proc/call_droppod), rand(100,300).` and spawn pod in `/obj/structure/strangebeacon/pods/proc/call_droppod()`